app/services/oneoffs/npq/create_new_contract_and_update_existing_statements.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+module Oneoffs::NPQ
+  class CreateNewContractAndUpdateExistingStatements
+    def initialize(path_to_csv:, cohort_year:, payment_date_range:)
+      @path_to_csv = path_to_csv
+      @cohort_year = cohort_year
+      @payment_date_range = payment_date_range
+    end
+
+    def call
+      ActiveRecord::Base.transaction do
+        rows_by_provider.each do |provider_name, rows|
+          cpd_lead_provider = CpdLeadProvider.find_by!(name: provider_name)
+          npq_lead_provider = cpd_lead_provider.npq_lead_provider
+
+          rows.each.with_index(1) do |row, index|
+            cohort = Cohort.find_by!(start_year: row["cohort_year"])
+            course = NPQCourse.find_by!(identifier: row["course_identifier"])
+
+            current_version = Finance::Statement::NPQ.where(cohort:, cpd_lead_provider:).pluck(:contract_version).max
+            new_version = Finance::ECF::ContractVersion.new(current_version).increment!
+
+            new_contract = NPQContract.find_or_initialize_by(
```

Comment:
I think we are missing a step which checks that all the previous contracts have either been recreated with the CSV updates or otherwise we'll need to duplicate and add the incremented version